### ISSUE TITLE
MongoDB and Mongoose: Create a Model

### ISSUE THREAD

**Issue by lasjorg** [freeCodeCamp/freeCodeCamp#49500]
### Describe the Issue

The challenge does not show the Mongoose schema syntax needed for an array of strings `property: [String]` instead it shows this `favoriteFoods : array of strings (*)`

Not sure why, but either the schema shown should show the actual syntax, or if that is considered a spoiler, an unrelated example should be given.

Also, note that the article linked does not show the syntax either. It does link to the Mongoose site at the end but in an unhelpful way. The article should have a link to the [schemas](https://mongoosejs.com/docs/guide.html#schemas) docs when introducing it and the [schematypes](https://mongoosejs.com/docs/schematypes.html).

### Affected Page

https://www.freecodecamp.org/learn/back-end-development-and-apis/mongodb-and-mongoose/create-a-model

### Your code

```
const schema = new mongoose.Schema({
  someRequiredString: { type: String, required: true },
  someNumber: Number,
  anArrayOfStrings: [String]
});
```

### Expected behavior

Update the schema shown or give an example that has the syntax.

### System

- Device: [e.g. iPhone 6, Laptop]
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.04]
- Browser: [e.g. Chrome, Safari]
- Version: [e.g. 22]

### Additional context

Forum: https://forum.freecodecamp.org/t/mongodb-and-mongoose-create-and-save-a-record-of-a-model/594436

**Comment by ShaunSHamilton** [comment]
I could go either way with this. It is not a common point of confusion (I assume most Campers just research it).

Personally, I would remove the table. I have never seen that pseudo-code syntax anywhere else. Also, I would link to the docs directly (but we try not to do this).

So, I suggest the article be updated to include a clear example of each common Mongoose type. @lasjorg Do you think that would be enough?

**Comment by lasjorg** [comment]
I think both should be updated.

Half the text in that step is about something unrelated to Schema and Schema types, it barely even explains what they are and how you use them. Linking to articles and documentation should be used to give more information, not for teaching what is expected knowledge to pass a step.

- The challenge should teach what it expects the students to know. If not thoroughly, then superficially.
- The article, if possible, should be updated with better documentation of the schema types and have links to the actual Mongoose docs where appropriate.

**Comment by Vansh-Baghel** [comment]
> I think both should be updated.
> 
> Half the text in that step is about something unrelated to Schema and Schema types, it barely even explains what they are and how you use them. Linking to articles and documentation should be used to give more information, not for teaching what is expected knowledge to pass a step.
> 
> * The challenge should teach what it expects the students to know. If not thoroughly, then superficially.
> * The article, if possible, should be updated with better documentation of the schema types and have links to the actual Mongoose docs where appropriate.

Agreed. So lets update the block 
This : 
`name : string [required],
age :  number,
favoriteFoods : array of strings (*)`
To this :
`name : {type: String, required: true},
age :  Number,
favoriteFoods : [String]`

And providing [[SchemaTypes link](https://mongoosejs.com/docs/schematypes.html#schematypes)](https://mongoosejs.com/docs/schematypes.html) would be better to explore the syntax and way of representation.

**Comment by Vansh-Baghel** [comment]
Do let me know if this is what is needed. Will work on it and resolve.

**Comment by ShaunSHamilton** [comment]
An issue with the https://github.com/freeCodeCamp/freeCodeCamp/labels/first%20timers%20only label is open for contribution. The first comprehensive PR created will be reviewed and merged. We typically do not assign issues to anyone other than long-time contributors.

If you would like to contribute, and have not read the contributors docs, please do so here: https://contribute.freecodecamp.org/#/

If you have any issues with contributing, be sure to join us on the [contributors channel](https://discord.com/channels/692816967895220344/715074489422970962), or on the [contributors sub-forum](https://forum.freecodecamp.org/c/contributors/3)

---

**Affected File**: https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/05-back-end-development-and-apis/mongodb-and-mongoose/create-a-model.md

**Tasks to Close**:

- [ ] Change instructions to:

```markdown
Create a person schema called `personSchema` with the following shape:

* A required `name` field of type `String`
* An `age` field of type `Number`
* A `favouriteFoods` field of type `[String]`

Use the Mongoose basic schema types...
```

- [ ] Update article to: **NOTE: Can only be done by staff**
  - Include a link to Mongoose schema docs
  - Show concrete examples of using the schema API


---

There is too much information in this lesson to add sufficient information within it about schema types. Ideally, this would be split into multiple lessons, but this is becoming legacy, and is not a problematic lesson - stick with minor tweaks.

**Comment by RandellDawson** [comment]
I left [this comment](https://github.com/freeCodeCamp/freeCodeCamp/pull/49542#issuecomment-1451381644) on the PR that was merged.

**Comment by ShaunSHamilton** [comment]
> I left [this comment](https://github.com/freeCodeCamp/freeCodeCamp/pull/49542#issuecomment-1451381644) on the PR that was merged.

Thank you for catching/mentioning that, @RandellDawson .

**Comment by scissorsneedfoodtoo** [comment]
Thanks everyone for your help with this.

I just updated the article that's linked in the challenge to include links to the Mongoose docs for `schema` and `SchemaTypes`. I also included a small example near the top to show how defining a schema works, a couple of schema types, and how to make a field required.

Those updates should go live soon within the next couple of hours.

It looks like the challenge has already been updated, so we should be able to close this issue once the updated article is live.
